conans/client/settings_preprocessor.py

```diff
@@ -0,0 +1,19 @@
+from conans.errors import InvalidConanSettingField
+from conans.util.log import logger
+
+
+def preprocess(settings):
+    fill_runtime(settings)
+
+
+def fill_runtime(settings):
+    try:
+        if settings.compiler == "Visual Studio":
+            if settings.get_safe("compiler.runtime") is None:
+                settings.compiler.runtime = "MDd" if settings.get_safe("build_type") == "Debug" \
+                                                  else "MD"
+                logger.info("Setting 'compiler.runtime' not declared, automatically "
+                            "adjusted to '%s'" % settings.compiler.runtime)
+    except InvalidConanSettingField:  # If the settings structure doesn't match these general
```

Comment:
If the user has changed the settings.yml defaults, it can raise a "BadValue" exception, not a InvalidConanSettingField, and that would be breaking and blocking.

I think this "smart" setting should catch any exception to make sure, so InvalidConanSettingField might be unnecessary.